src/server/game/Movement/MovementGenerators/ConfusedMovementGenerator.cpp

```diff
@@ -56,44 +56,38 @@ bool ConfusedMovementGenerator<T>::DoUpdate(T* unit, uint32 diff)
     if (unit->HasUnitState(UNIT_STATE_ROOT | UNIT_STATE_STUNNED | UNIT_STATE_DISTRACTED))
         return true;
 
-    if (i_nextMoveTime.Passed())
-    {
-        // currently moving, update location
-        unit->AddUnitState(UNIT_STATE_CONFUSED_MOVE);
```

Comment:
what happend with this unit state?

Reply:
Isn't adding it on init and reset enough? Or does something remove it?

Reply:
It doesn't seem anything in core would clear all unit states or specifically confused move - only confused movement generator itself removes it